src/main/resources/include/hail/NativeStatus.h

```diff
@@ -0,0 +1,47 @@
+#ifndef HAIL_NATIVESTATUS_H
+#define HAIL_NATIVESTATUS_H 1
+
+#include "hail/NativeObj.h"
+#include <cstdarg>
+#include <cstdio>
+#include <string>
+
+namespace hail {
+
+class NativeStatus : public NativeObj {
```

Comment:
This doesn't override `get_class_name`.  In fact, none of the classes that extend NativeObj do.  We should define it or delete it.